code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm

```diff
@@ -132,57 +133,52 @@
 	Coatings have both a raw and a cooked image. Raw coating is generally unhealthy
 	Generally coatings are intended for deep frying foods
 */
-/datum/reagent/nutriment/coating
+/decl/reagent/nutriment/coating
 	nutriment_factor = 6 //Less dense than the food itself, but coatings still add extra calories
-	var/messaged = 0
 	var/icon_raw
 	var/icon_cooked
 	var/coated_adj = "coated"
 	var/cooked_name = "coating"
 	taste_description = "some sort of frying coating"
 
-/datum/reagent/nutriment/coating/affect_ingest(var/mob/living/carbon/M, var/alien, var/removed)
+/decl/reagent/nutriment/coating/initial_effect(mob/living/carbon/M, alien, datum/reagents/holder)
+	. = ..()
+	to_chat(M, "Ugh, this raw [name] tastes disgusting.")
 
-	//We'll assume that the batter isnt going to be regurgitated and eaten by someone else. Only show this once
-	if (data["cooked"] != 1)
-		if (!messaged)
-			to_chat(M, "Ugh, this raw [name] tastes disgusting.")
-			nutriment_factor *= 0.5
-			messaged = 1
+/decl/reagent/nutriment/coating/digest(var/mob/living/carbon/M, var/removed, var/datum/reagents/holder)
+	var/nut_fact = holder.reagent_data[type]["cooked"] == TRUE ? nutriment_factor : nutriment_factor / 2 // it's the nut fact
+	M.heal_organ_damage(regen_factor * removed, 0)
+	M.adjustNutritionLoss(-nut_fact * removed)
+	M.nutrition_attrition_rate = Clamp(M.nutrition_attrition_rate + attrition_factor, 1, 2)
+	M.add_chemical_effect(CE_BLOODRESTORE, blood_factor * removed)
+	M.intoxication -= min(M.intoxication,nut_fact*removed*0.05) //Nutrients can absorb alcohol.
 
+/decl/reagent/nutriment/coating/affect_ingest(var/mob/living/carbon/M, var/alien, var/removed, var/datum/reagents/holder)
+	if (holder.reagent_data[type]["cooked"] != 1)
```

Comment:
```suggestion
	if (!holder.reagent_data[type]["cooked"])
```

Reply:
Nnnnnnnnnno, cooked can be -1 under some circumstances, iirc?

Reply:
Always TRUE/FALSE as discussed on Discord.